File: algorithms/1317.convert-integer-to-the-sum-of-two-no-zero-integers/solution.go
func getNoZeroIntegers(n int) []int {
	var res []int

	for i := 1; i < n; i++ {
		if noZero(i) && noZero(n-i) {
			res = []int{i, n - i}
			break
		}
	}

	return res
}

func noZero(num int) bool {
	for num > 0 {
		if num%10 == 0 {
			return false
		}

		num /= 10
	}

	return true
}

File: algorithms/1317.convert-integer-to-the-sum-of-two-no-zero-integers/solution.rb
# @param {Integer} n
# @return {Integer[]}
def get_no_zero_integers(n)
  (1...n).each do |i|
    if no_zero?(i) && no_zero?(n - i)
      return [i, n - i]
    end
  end
end

def no_zero?(num)
  while num > 0
    return false if num % 10 == 0

    num /= 10
  end

  true
end
